OpenRA.Test/OpenRA.Game/ActorInfoTest.cs

```diff
@@ -88,5 +92,65 @@ public void TraitsInConstructOrderC()
 				Assert.That(count, Is.EqualTo(Math.Floor(count)), "Should be symmetrical");
 			}
 		}
+
+		[TestCase(TestName = "Trait inheritance and removal can be composed")]
+		public void TraitInheritanceAndRemovalCanBeComposed()
+		{
+			var baseYaml = @"
+^BaseA:
+	MockA2:
+^BaseB:
+	Inherits@a: ^BaseA
+	MockB2:
+";
+			var extendedYaml = @"
+Actor:
+	Inherits@b: ^BaseB
+	-MockA2:
+";
+			var mapYaml = @"
+^BaseC:
+	MockC2:
+Actor:
+	Inherits@c: ^BaseC
+";
+
+			var actorInfo = CreateActorInfoFromYaml("Actor", mapYaml, baseYaml, extendedYaml);
+			Assert.IsFalse(actorInfo.HasTraitInfo<MockA2Info>(), "Actor should not have the MockA2 trait, but does.");
+			Assert.IsTrue(actorInfo.HasTraitInfo<MockB2Info>(), "Actor should have the MockB2 trait, but does not.");
+			Assert.IsTrue(actorInfo.HasTraitInfo<MockC2Info>(), "Actor should have the MockC2 trait, but does not.");
+		}
+
+		[TestCase(TestName = "Trait can be removed after multiple inheritance")]
+		public void TraitCanBeRemovedAfterMultipleInheritance()
+		{
+			var baseYaml = @"
+^BaseA:
+	MockA2:
+Actor:
+	Inherits: ^BaseA
+	MockA2:
+";
+			var overrideYaml = @"
+Actor:
+	-MockA2
+";
+
+			var actorInfo = CreateActorInfoFromYaml("Actor", null, baseYaml, overrideYaml);
+			Assert.IsFalse(actorInfo.HasTraitInfo<MockA2Info>(), "Actor should not have the MockA2 trait, but does.");
+		}
+
+		// This needs to match the logic used in RulesetCache.LoadYamlRules
```

Comment:
My expectation is that it wouldn't bitrot fast. This part of the code is not supposed to change often anymore (as compared to, say, six years ago). And if it does, chances are that the test would fail. Also the code isn't copied verbatim, but just does the same thing in spirit, so to speak, so reusing it would require rewriting the code in `RulesetCache.LoadYamlRules` to make it reusable first. And that would be too risky for too little gain, IMO.